### PR TITLE
[BABEL-2902] Expand explain for PRINT, THROW, USE and RAISERROR

### PR DESCRIPTION
**Pull request by jusjosj**
This PR adds support to babelfish_showplan_all for print, throw, raiserror, and use <database>

### Description

This PR extends the functionality of babelfish_showplan_all to include print, throw, raiserror, and use <database>
This is largely done in a manner consistent with the other explain functions existing throughout the codebase.  

 
### Issues Resolved

BABEL-2902

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 1]
```diff
@@ -99,44 +99,75 @@ static int exec_stmt_raiserror(PLtsql_execstate *estate, PLtsql_stmt_raiserror *
 	int		severity = -1;
 	int		state = -1;
 
-	/* Read parameters of RAISERROR statements */
-	read_raiserror_params(estate, stmt->params, stmt->paramno, &msg, &msg_id, &severity, &state);
-	msg = pstrdup(msg);
+	PLtsql_expr *temp_expr;
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c]: Let's create another function to handle `pltsql_explain_only` and call it here. In that way, we can make the original function simple and reduce unnecessary diffs.

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 2]
```diff
@@ -99,44 +99,75 @@ static int exec_stmt_raiserror(PLtsql_execstate *estate, PLtsql_stmt_raiserror *
...
+	if (pltsql_explain_only) {
+		/* calculate stmt length */
+		stmt_len = 0;
+		for (int i = 0; i < stmt->paramno; i++)
+		{
+			temp_expr = (PLtsql_expr *) list_nth(stmt->params, i);
+			/* minus 6 is stripping 'SELECT ' and adding one space for a comma */
+			stmt_len += strlen(temp_expr->query) - 6;
+		}
+		char *query_text = (char *) palloc0(sizeof(char) * stmt_len);
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c]: Instead of using `palloc` and `strncat`, let's try to use `StringInfoData` if possible.

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 3]
```diff
@@ -635,21 +697,9 @@ static inline int dispatch_stmt(PLtsql_execstate *estate, PLtsql_stmt *stmt)
 			exec_stmt_close(estate, (PLtsql_stmt_close *) stmt);
 			break;
 		case PLTSQL_STMT_COMMIT:
-			if (pltsql_explain_only)
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c]: Let's keep this until we remove all `PLTSQL_STMT_COMMIT`-related code.

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 4]
```diff
@@ -147,35 +178,66 @@ static int exec_stmt_throw(PLtsql_execstate *estate, PLtsql_stmt_throw *stmt)
...
 		/* Check if we are inside a CATCH block */
 		if (estate->cur_error == NULL || estate->cur_error->error == NULL)
 			ereport(ERROR, (errcode(ERRCODE_INTERNAL_ERROR),
 					errmsg("THROW without parameters should be executed inside a CATCH block")));
-
+		/* If only explaining, don't actually perform the throw, just append query text */
+		if (pltsql_explain_only)
+		{
+			append_explain_info(NULL, "THROW");
+		}
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c:167]: Can we add `return PLTSQL_RC_OK;` here so that we don't need `else`?

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 5]
```diff
@@ -147,35 +178,66 @@ static int exec_stmt_throw(PLtsql_execstate *estate, PLtsql_stmt_throw *stmt)
...
 		ReThrowError(estate->cur_error->error);
+		}
 	}
 	else
 	{
 		/* Read parameters of THROW statement */
-		read_throw_params(estate, stmt->params, &msg, &err_no, &state);
-		msg = pstrdup(msg);
-		proc_name = get_proc_name(estate);
+		if (pltsql_explain_only)
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c:179]: It would be nice to have a separate function.

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 6]
```diff
@@ -1542,6 +1571,17 @@ void process_explain(PLtsql_execstate *estate)
 	if (*pltsql_protocol_plugin_ptr && (*pltsql_protocol_plugin_ptr)->stmt_beg)
 		((*pltsql_protocol_plugin_ptr)->stmt_beg) (estate, NULL);
 
+	/* If use_db changed db during the query, return it back to the starting database */
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c:1534]: We should not allow to change db. Let's think about the way to prevent this instead of returning back to the starting db.

File: contrib/babelfishpg_tsql/src/pl_exec-2.c [hunk 7]
```diff
@@ -233,29 +234,37 @@ exec_stmt_print(PLtsql_execstate *estate, PLtsql_stmt_print *stmt)
...
-	formatdatum = exec_eval_expr(estate,
-								 (PLtsql_expr *) linitial(stmt->exprs),
-								 &formatisnull,
-								 &formattypeid,
-								 &formattypmod);
-
-	if (formatisnull)
-		extval = "<NULL>";
+
+	 if (pltsql_explain_only) {
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_exec-2.c]: Same as the above comments:
- remove `else`
- use `StringInfoData`

File: contrib/babelfishpg_tsql/src/pl_exec-2.c [hunk 8]
```diff
@@ -558,7 +567,7 @@ exec_stmt_push_result(PLtsql_execstate *estate,
 					 receiver,
 					 receiver,
 					 &qc))
-                processed = portal->portalPos;
+				processed = portal->portalPos;
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_exec-2.c]: indent?

File: contrib/babelfishpg_tsql/src/pl_exec-2.c [hunk 9]
```diff
@@ -569,7 +578,7 @@ exec_stmt_push_result(PLtsql_execstate *estate,
 
 	estate->eval_processed = processed;
 	exec_set_rowcount(processed);
-        exec_set_found(estate, processed != 0);
+	exec_set_found(estate, processed != 0);
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_exec-2.c]: Let't try to minimize unnecessary changes.

File: contrib/babelfishpg_tsql/src/pl_exec.c [hunk 10]
```diff
@@ -6081,13 +6081,18 @@ exec_stmt_close(PLtsql_execstate *estate, PLtsql_stmt_close *stmt)
 static int
 exec_stmt_commit(PLtsql_execstate *estate, PLtsql_stmt_commit *stmt)
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_exec.c]: Do we need any changes in `exec_stmt_commit`?

File: contrib/babelfishpg_tsql/src/pltsql.h [hunk 11]
```diff
@@ -1280,6 +1280,9 @@ typedef struct ExplainInfo
 
 	/* indent for the next ExplainInfo */
 	size_t next_indent;
+
+	/* used to restore session to original schema if "use db" is invoked */
+    char *initial_schema;
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pltsql.h]: I think `initial_database` would be the better name.

File: test/JDBC/input/babel_2902.sql [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+CREATE TABLE ma1 (a int)
+GO
+
+-- CREATE DATABASE testdb1
+-- GO
+-- USE testdb1
+-- CREATE TABLE t1 (z int)
+-- GO
+
+-- CREATE DATABASE testdb2
```
Comment (2jungkook) [inline test/JDBC/input/babel_2902.sql]: You may need multidb settings.

File: contrib/babelfishpg_tsql/src/pl_explain.c [hunk 13]
```diff
@@ -170,4 +170,19 @@ void append_explain_info(QueryDesc *queryDesc, const char *queryString)
 
 	/* Recover the memory context */
 	MemoryContextSwitchTo(oldcxt);
+}
+
+void set_explain_schema(char *db_name)
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_explain.c]: Let's use `const char *` if it is not modified.

File: contrib/babelfishpg_tsql/src/iterative_exec.c (hunk 6 above)
Comment (jusjosj) [reply contrib/babelfishpg_tsql/src/iterative_exec.c:1534]: I think that would be a pretty substantial departure from SQL Server.  SQL Server seems to internally process use db.  Here's a snippet showing use db from sql server

```
    1> create database showplan
    2> go
    1> use showplan
    2> create table test_showplan_all(a int)
    3> go
    Changed database context to 'showplan'.
    1> use master
    2> go
    Changed database context to 'master'.
    1> set showplan_all on
    2> go
    1> use showplan
    2> select * from test_showplan_all
    3> go
    StmtText                                                        StmtId      NodeId      Parent      PhysicalOp                     LogicalOp                      Argument                                       DefinedValues                             EstimateRows   EstimateIO     EstimateCPU    AvgRowSize  TotalSubtreeCost OutputList                                Warnings Type                                                             Parallel EstimateExecutions
    --------------------------------------------------------------- ----------- ----------- ----------- ------------------------------ ------------------------------ ---------------------------------------------- 
    ----------------------------------------- -------------- -------------- -------------- ----------- ---------------- ----------------------------------------- -------- ---------------------------------------------------------------- -------- ------------------
    use showplan                                                              1           1           0 NULL                           NULL                           1                                              NULL                                                NULL           NULL           NULL        NULL             NULL NULL                                      NULL     USE DATABASE                                                            0               NULL
    select * from test_showplan_all                                           2           2           0 NULL                           NULL                           2                                              NULL                                                 1.0           NULL           NULL        NULL        0.0032831 NULL                                      NULL     SELECT                                                                  0               NULL
      |--Table Scan(OBJECT:([showplan].[dbo].[test_showplan_all]))            2           3           2 Table Scan                     Table Scan                     OBJECT:([showplan].[dbo].[test_showplan_all])  [showplan].[dbo].[test_showplan_all].[a]             1.0   3.2035001E-3   7.9600002E-5          11        0.0032831 [showplan].[dbo].[test_showplan_all].[a]  NULL     PLAN_ROW                                                                0                1.0
     
    (3 rows affected)
    1> select * from test_showplan_all
    2> go
    Msg 208, Level 16, State 1, Server EC2AMAZ-536AJ70, Line 1
    Invalid object name 'test_showplan_all'.
````

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 14]
```diff
@@ -738,31 +782,16 @@ static inline int dispatch_stmt(PLtsql_execstate *estate, PLtsql_stmt *stmt)
...
 			break;
 		case PLTSQL_STMT_USEDB:
 			if (pltsql_explain_only)
-			{
-				ereport(ERROR,
-						(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						errmsg("Showing Estimated Execution Plan for USE DB statment is not yet supported")));
-			}
-			exec_stmt_usedb(estate, (PLtsql_stmt_usedb *) stmt);
+				exec_stmt_usedb_explain(estate, (PLtsql_stmt_usedb *) stmt, false);
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c]: Let's call `exec_stmt_usedb_explain()` inside `exec_stmt_usedb()`.

File: contrib/babelfishpg_tsql/src/pl_exec-2.c [hunk 15]
```diff
@@ -2536,27 +2545,88 @@ exec_stmt_usedb(PLtsql_execstate *estate, PLtsql_stmt_usedb *stmt)
...
 		((*pltsql_protocol_plugin_ptr)->send_info) (0, 1, 0, message, 0);
+	
+	return PLTSQL_RC_OK;
+}
+
+static int
+exec_stmt_usedb_explain(PLtsql_execstate *estate, PLtsql_stmt_usedb *stmt, bool shouldRestoreDb)
+{
+	/* This function will change databases to a given target database for use in explain functions
+	 * It will maintain the lock on the initial database and supress any log messages to the user
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_exec-2.c]: Why do we need this?

File: contrib/babelfishpg_tsql/src/pl_exec.c [hunk 16]
```diff
@@ -7527,10 +7532,12 @@ exec_eval_simple_expr(PLtsql_execstate *estate,
 
 	/*
 	 * Finally we can call the executor to evaluate the expression
+	 *  if not in explain only mode
 	 */
-	*result = ExecEvalExpr(expr->expr_simple_state,
-						   econtext,
-						   isNull);
+	if (!pltsql_explain_only)
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_exec.c]: When do we need this?

File: contrib/babelfishpg_tsql/src/pl_exec-2.c (hunk 8 above)
Comment (jusjosj) [reply contrib/babelfishpg_tsql/src/pl_exec-2.c]: I think this was an auto indent change from tabs to spaces.  Its indented by 4 tabs, arguments to the if are 5 tabs.  I'm not sure what are convention is when there's this many tabs.  I can change this line back to spaces if that's preferable.

File: contrib/babelfishpg_tsql/src/pl_exec.c (hunk 10 above)
Comment (jusjosj) [reply contrib/babelfishpg_tsql/src/pl_exec.c]: No will revert

File: contrib/babelfishpg_tsql/src/pl_exec.c (hunk 16 above)
Comment (jusjosj) [reply contrib/babelfishpg_tsql/src/pl_exec.c]: this is not needed any longer.  Its an artifact of trying to get explain to work with function calls

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 17]
```diff
@@ -99,6 +101,11 @@ static int exec_stmt_raiserror(PLtsql_execstate *estate, PLtsql_stmt_raiserror *
 	int		severity = -1;
 	int		state = -1;
 
+	if (pltsql_explain_only) {
+
+		read_raiserror_params_explain(stmt->params, stmt->paramno);
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c]: nit; we can make `read_raiserror_params_explain` function to return PLTSQL_RC_OK.

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 18]
```diff
@@ -154,18 +160,28 @@ static int exec_stmt_throw(PLtsql_execstate *estate, PLtsql_stmt_throw *stmt)
...
-							estate->cur_error->severity,
-							estate->cur_error->state,
-							true /* rethrow */);
-		ReThrowError(estate->cur_error->error);
+	set_exec_error_data(estate->cur_error->procedure,
+						estate->cur_error->number,
+						estate->cur_error->severity,
+						estate->cur_error->state,
+						true /* rethrow */);
+	ReThrowError(estate->cur_error->error);
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c]: wrong indent

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 19]
```diff
@@ -1542,6 +1534,17 @@ void process_explain(PLtsql_execstate *estate)
...
 
+	/* If use_db changed db during the query, return it back to the starting database */
+	if (estate->explain_infos)
+	{
+		einfo = (ExplainInfo *) llast(estate->explain_infos);
+		if (einfo->initial_database)
+		{
+			PLtsql_stmt_usedb *initial_db = palloc0(sizeof(PLtsql_stmt_usedb));
+			initial_db->db_name = einfo->initial_database;
+			exec_stmt_usedb_explain(estate, (PLtsql_stmt_usedb *) initial_db, true);
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c]: casting is unnecessary

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 20]
```diff
@@ -1685,7 +1688,29 @@ void assert_equal_estate_err(PLtsql_estate_err *err1, PLtsql_estate_err *err2)
...
+	
+	if (!pltsql_explain_only)
+		return;
 
+	initStringInfo(&query_string);
+	appendStringInfo(&query_string, "RAISERROR (");
+	for (int i = 0; i < paramno; i++)
+	{
+		exprTemp = (PLtsql_expr *) list_nth(params, i);
+		appendStringInfoString(&query_string, &exprTemp->query[7]);
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c]: 1. Add a null check for exprTemp
2. Add a comment explaining why we use `7`
3. Add a string length check before accessing `query[7]`

File: contrib/babelfishpg_tsql/src/iterative_exec.c [hunk 21]
```diff
@@ -1760,6 +1785,27 @@ static void read_raiserror_params(PLtsql_execstate *estate, List *params, int pa
...
+	exprTemp = (PLtsql_expr *) list_nth(params, 0);
+	appendStringInfoString(&query_text, &exprTemp->query[7]);
+	appendStringInfoString(&query_text, ", ");
+
+	exprTemp = (PLtsql_expr *) list_nth(params, 1);
+	appendStringInfoString(&query_text, &exprTemp->query[7]);
+	appendStringInfoString(&query_text, ", ");
+
+	exprTemp = (PLtsql_expr *) list_nth(params, 2);
+	appendStringInfoString(&query_text, &exprTemp->query[7]);
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/iterative_exec.c]: Same as the above.
1. list length check
2. comments regarding `7`
3. string length check

File: contrib/babelfishpg_tsql/src/pl_exec-2.c [hunk 22]
```diff
@@ -233,20 +234,29 @@ exec_stmt_print(PLtsql_execstate *estate, PLtsql_stmt_print *stmt)
...
 		extval = "<NULL>";
 	else
 		extval = convert_value_to_string(estate,
-										 formatdatum,
-										 formattypeid);
+										formatdatum,
+										formattypeid);
 
-	ereport(INFO, errmsg_internal("%s", extval));
+ 	ereport(INFO, errmsg_internal("%s", extval));
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_exec-2.c]: There are still many unnecessary code changes. Please fix them all.

File: contrib/babelfishpg_tsql/src/pl_exec-2.c [hunk 23]
```diff
@@ -2557,6 +2569,67 @@ exec_stmt_usedb(PLtsql_execstate *estate, PLtsql_stmt_usedb *stmt)
...
+	/* This function will change databases to a given target database for use in explain functions
+	 * It will maintain the lock on the initial database and supress any log messages to the user
+	 * otherwise this function will be functionally the same as exec_stmt_usedb
+	 */
+	if (!pltsql_explain_only)
+		return PLTSQL_RC_OK;
+
+	char *old_db_name;
+	char *initial_database_name;
+	char message[128];
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_exec-2.c]: not used

File: contrib/babelfishpg_tsql/src/pl_exec-2.c [hunk 24]
```diff
@@ -2513,6 +2522,9 @@ static Node *get_underlying_node_from_implicit_casting(Node *n, NodeTag underlyi
 static int
 exec_stmt_usedb(PLtsql_execstate *estate, PLtsql_stmt_usedb *stmt)
 {
+	if (pltsql_explain_only) {
+		return exec_stmt_usedb_explain(estate, stmt, false);
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_exec-2.c]: nit; we can use `..., stmt, false /* shouldRestoreDb */);` for a better readability.

File: contrib/babelfishpg_tsql/src/pl_exec-2.c [hunk 25]
```diff
@@ -2557,6 +2569,67 @@ exec_stmt_usedb(PLtsql_execstate *estate, PLtsql_stmt_usedb *stmt)
...
 	if (*pltsql_protocol_plugin_ptr && (*pltsql_protocol_plugin_ptr)->send_info)
 		((*pltsql_protocol_plugin_ptr)->send_info) (0, 1, 0, message, 0);
+	
+	return PLTSQL_RC_OK;
+}
+
+static int
+exec_stmt_usedb_explain(PLtsql_execstate *estate, PLtsql_stmt_usedb *stmt, bool shouldRestoreDb)
+{
+	/* This function will change databases to a given target database for use in explain functions
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_exec-2.c]: nit; usually, we put the this kind of comment outside of a function.

File: contrib/babelfishpg_tsql/src/pl_exec-2.c [hunk 26]
```diff
@@ -2557,6 +2569,67 @@ exec_stmt_usedb(PLtsql_execstate *estate, PLtsql_stmt_usedb *stmt)
...
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_DATABASE),
+				 errmsg("database \"%s\" does not exist", stmt->db_name)));
+		
+	}
+	/* Release the session-level shared lock on the old logical db if its not the user's original database */
+	if (old_db_id != initial_database_id)
+		UnlockLogicalDatabaseForSession(old_db_id, ShareLock, false);
+
+	/* Get a session-level shared lock on the new logical db we are about to use. If initial db then no need to relock since we still have one */
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_exec-2.c]: `If initial db then ...` --> `when we restoring the initial db ...`

File: contrib/babelfishpg_tsql/src/pl_explain.c [hunk 27]
```diff
@@ -165,9 +168,25 @@ void append_explain_info(QueryDesc *queryDesc, const char *queryString)
...
+{
+       ExplainInfo *einfo = get_last_explain_info();
+       einfo->initial_database = db_name;
+}
+
+char *get_explain_database(void)
+{
+       ExplainInfo *einfo = get_last_explain_info();
+       if (einfo != NULL && einfo->initial_database)
+               return einfo->initial_database;
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pl_explain.c]: This is equivalent to
```
if (einfo)
    return einfo->initial_database;
```

File: contrib/babelfishpg_tsql/src/pltsql.h [hunk 28]
```diff
@@ -1280,6 +1280,9 @@ typedef struct ExplainInfo
 
 	/* indent for the next ExplainInfo */
 	size_t next_indent;
+
+	/* used to restore session to original schema if "use db" is invoked */
+    char *initial_database;
```
Comment (2jungkook) [inline contrib/babelfishpg_tsql/src/pltsql.h]: Let's use `const char *` whenever possible.

File: test/JDBC/expected/BABEL-2902.out [hunk 29]
```diff
@@ -0,0 +1,548 @@
...
+    SET @a = @a + 1
+END
+GO
+~~START~~
+text
+Query Text: ASSIGN @a = SELECT 0
+  Query Text: SELECT 0
+  ->  Result  (cost=0.00..0.01 rows=1 width=4)
+Query Text: SELECT COUNT(*) FROM babel_2902_ma1
+Aggregate  (cost=41.88..41.89 rows=1 width=4)
```
Comment (2jungkook) [inline test/JDBC/expected/BABEL-2902.out]: cost may vary between versions. Let's use
```
select set_config('babelfishpg_tsql.explain_costs', 'off', false);
```

File: test/JDBC/expected/BABEL-2902.out [hunk 30]
```diff
@@ -0,0 +1,548 @@
...
+GO
+~~START~~
+nvarchar
+babel_2902_testdb_1
+~~END~~
+
+
+
+---- TEST RAISEERROR ----
+-- note that during explain SQL Server does very limited no validation on inputs to raiserror
```
Comment (2jungkook) [inline test/JDBC/expected/BABEL-2902.out]: Let's use T-SQL instead of SQL Server.

File: test/JDBC/expected/BABEL-2902.out [hunk 31]
```diff
@@ -0,0 +1,548 @@
...
+
+-- try catch --
+THROW
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: THROW without parameters should be executed inside a CATCH block)~~
+
+
+-- Unable to test until explain for RESTORE CTX PARTIAL is implemented
```
Comment (2jungkook) [inline test/JDBC/expected/BABEL-2902.out:522]: Can we uncomment the below? It's worth to add a test case even throwing unsupported errors because it can guide other developers when they implement a new feature.

File: contrib/babelfishpg_tsql/src/iterative_exec.c (hunk 20 above)
Comment (jusjosj) [reply contrib/babelfishpg_tsql/src/iterative_exec.c]: Would you like me to add similar null checks in the main functions as well?

**Review by 2jungkook — APPROVED** [review]

**Comment by 2jungkook** [comment]
Waiting for an internal Code Review.
